src/Apps/Artwork/Components/OtherWorks/index.tsx

```diff
@@ -0,0 +1,126 @@
+import { Spacer } from "@artsy/palette"
+import { OtherWorks_artwork } from "__generated__/OtherWorks_artwork.graphql"
+import { OtherWorksQuery } from "__generated__/OtherWorksQuery.graphql"
+import { ContextConsumer } from "Artsy"
+import { renderWithLoadProgress } from "Artsy/Relay/renderWithLoadProgress"
+import React from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { OtherWorksByArtistFragmentContainer as OtherWorksByArtist } from "./OtherWorksByArtist"
+
+interface OtherWorksProps {
+  artwork: OtherWorks_artwork
+}
+
+export const OtherWorks: React.SFC<OtherWorksProps> = props => {
+  const grids = getGridsByArtworkType(props)
+
+  return (
+    <>
+      {grids.map((GridComponent, index) => {
+        const addSpacer = index !== grids.length - 1
+
+        return (
+          <>
+            <GridComponent artwork={props.artwork} key={index} />
+            {addSpacer && <Spacer mb={2} />}
+          </>
+        )
+      })}
+    </>
+  )
+}
+
+/*
+  TODO: Grid Order by type:
+
+  ArtworkContextAuction:
+    - PartnerArtworks
+    - AuctionArtworks @skip(if: $isClosed)
+    - ArtistArtworks @include(if: $isClosed)
+    - RelatedArtworks @include(if: $isClosed)
+
+  ArtworkContextFair:
+    - FairArtworks
+    - PartnerArtworks
+    - ArtistArtworks
+    - RelatedArtworks
+
+  ArtworkContextShow:
+    - ShowArtworks
+    - ArtistArtworks
+    - PartnerArtworks
+    - RelatedArtworks
+
+  Default:
+    - ArtistArtworks
+    - PartnerArtworks
+    - RelatedArtworks
+*/
+
+const getGridsByArtworkType = props => {
+  const artworkContextType = props.artwork.context.__typename
+
+  switch (artworkContextType) {
+    case "ArtworkContextAuction":
+      return [OtherWorksByArtist, OtherWorksByArtist]
+    case "ArtworkContextFair":
+      return [OtherWorksByArtist]
+    case "ArtworkContextPartnerShow":
+      return [OtherWorksByArtist]
+    default:
+      return [OtherWorksByArtist]
+  }
+}
+
+export const OtherWorksFragmentContainer = createFragmentContainer(
+  OtherWorks,
+  graphql`
+    fragment OtherWorks_artwork on Artwork {
+      context {
+        __typename
+      }
+
+      ...OtherWorksByArtist_artwork
```

Comment:
I commented above in `ArtworkApp`, but basically, this component should wind up listing all the 'other works' grids, but they should be guarded appropriately behind the 'type' of work it is (so `@include(if: $shouldShowArtist)`).

Then, the query renderer for this below, can figure out the proper variables for that, based on incoming props: `{ artworkID: ..., artworkContext: ...}`

The types of grids to fetch for a given artwork 'context' should be the domain of this component, so the artwork 'context' should be passed in as a prop.